### PR TITLE
cli: implement building executables for all platforms

### PR DESCRIPTION
**Pull request by menuka94**
# Purpose
> Implement the functionality to build executables for all platforms (Mac, Linux, and Windows).

Related to https://github.com/wso2/micro-integrator/issues/139.

### CODE REVIEW

**Comment by a5anka** [comment]
Let's use a makefile as discussed offline

**Comment by menuka94** [comment]
Now it is possible to execute `make install` and build both the **Micro Integrator** and **MIcro Integrator CLI** at once.

File: cmd/build.sh [hunk 1]
```diff
@@ -1,47 +1,132 @@
...
+then
+    filename=${filename%.go}
+fi
+
+#platforms="darwin/amd64 freebsd/386 freebsd/amd64 freebsd/arm linux/386 linux/amd64 linux/arm windows/386 windows/amd64"
+#platforms="linux/amd64/linux/x64"
+#platforms="darwin/amd64/macosx/x64"
+if [ "${full_build}" == "true" ]; then
+# the following line give an error in MacOS
+#    echo "Building "$'\e[1m'"${filename^^}:${build_version}"$'\e[0m'" for all platforms..."
```
Comment (a5anka) [inline cmd/build.sh]: Can't we fix this?

Comment (menuka94) [reply cmd/build.sh]: Fixed.

**Pull request merged**